### PR TITLE
feature: Settings modal + quick select panel

### PR DESCRIPTION
**Pull request by timarney**
# Summary | Résumé

- Adds "quick select panel" for classification and branding
- Adds "...more" settings modal
- Updates publish page
- Updates Left nav to move settings up
- Updates settings page
- Updates branding page

For any UI changes please include screenshots at different breakpoints. It would also be helpful to include a
before and after comparison for each change i.e.

Screenshot 2023-10-10 at 1 36 49 PM

Screenshot 2023-10-10 at 1 38 20 PM

# Test instructions | Instructions pour tester la modification

- Visit edit form
1)  Use the quick select to change classification and branding settings

2) Use the settings modal to update classification and branding settings

3) Visit the publish page to -- note the last item indicating the classification and delivery type --- use change link to update.

# Pull Request Checklist

Please complete the following items in the checklist before you request a review:

- [x] Have you completely tested the functionality of change introduced in this PR? Is the PR solving the problem it's meant to solve within the scope of the related issue?
- [x] The PR does not introduce any new issues such as failed tests, console warnings or new bugs.
- [ ] If this PR adds a package have you ensured its licensed correctly and does not add additional security issues?
- [ ] Is the code clean, readable and maintainable? Is it easy to understand and comprehend.
- [ ] Does your code have adequate comprehensible comments? Do new functions have [docstrings](https://tsdoc.org/)?
- [ ] Have you modified the change log and updated any relevant documentation?
- [ ] Is there adequate test coverage? Both unit tests and end-to-end tests where applicable?
- [ ] If your PR is touching any UI is it accessible? Have you tested it with a screen reader? Have you tested it with automated testing tools such as axe?

### CODE REVIEW

File: components/form-builder/app/ClassificationSelect.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
+import React from "react";
+import { cn } from "@lib/utils";
+import { ProtectedIcon } from "@formbuilder/icons";
+
+const classificationOptions = [
+  { value: "Unclassified", en: "Unclassified", fr: "Non classifié" },
```
Comment (thiessenp-cds) [inline components/form-builder/app/ClassificationSelect.tsx:6]: Curious why a hardcoded string (vs. i18n)?

**Review by thiessenp-cds — COMMENTED** [review]
I tested the changes on OSX with VoiceOver and on Windows with NVDA. Nice use of labels etc. - works well. Will approve once out of draft :)

File: components/form-builder/app/ClassificationSelect.tsx (hunk 1 above)
Comment (timarney) [reply components/form-builder/app/ClassificationSelect.tsx:6]: Mostly a TypeScript thing so we can declare the en / fr keys outside of the component and not have them start as empty strings. 

Screenshot 2023-10-11 at 12 14 52 PM

Comment (anikbrazeau) [inline components/form-builder/app/ClassificationSelect.tsx:6]:
```suggestion
  { value: "Unclassified", en: "UNCLASSIFIED", fr: "NON CLASSIFIÉ" },
```

File: components/form-builder/app/ClassificationSelect.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  { value: "Protected A", en: "Protected A (default)", fr: "Protégé A (défaut)" },
+  { value: "Protected B", en: "Protected B", fr: "Protégé B" },
```
Comment (anikbrazeau) [inline components/form-builder/app/ClassificationSelect.tsx:8]:
```suggestion
  { value: "Protected B", en: "PROTECTED B", fr: "PROTÉGÉ B" },
```

File: public/static/locales/fr/form-builder.json [hunk 3]
```diff
@@ -652,5 +655,32 @@
       "body2": "réponses non confirmées avec 35 jours de retard ou plus. Confirmez la réception de toute réponse téléchargée en retard avant d'en enregistrer de nouvelles."
     },
     "loading": "Chargement..."
+  },
+  "formSettingsModal": {
+    "title": "Form set up [fr]",
```
Comment (anikbrazeau) [inline public/static/locales/fr/form-builder.json:660]:
```suggestion
    "title": "Configurer le formulaire",
```

File: public/static/locales/fr/form-builder.json [hunk 4]
```diff
@@ -652,5 +655,32 @@
...
+      "text1": "Save changes to form settings before closing? [fr]",
+      "text2": "Save or Cancel [fr]"
+    },
+    "more": "More [fr]"
+  },
+  "classificationSelect": {
+    "label": "Select the data classification of your form responses [fr]"
+  },
+  "brandingSelect": {
+    "label": "Select branding for your form [fr]"
```
Comment (anikbrazeau) [inline public/static/locales/fr/form-builder.json:684]:
```suggestion
    "label": "Sélectionnez l'image de marque de votre formulaire"
```

File: public/static/locales/fr/form-builder.json [hunk 5]
```diff
@@ -652,5 +655,32 @@
...
+      "text2": "Email delivery is not available for this classification [fr]"
+    },
+    "saveNote": {
+      "text1": "Save changes to form settings before closing? [fr]",
+      "text2": "Save or Cancel [fr]"
+    },
+    "more": "More [fr]"
+  },
+  "classificationSelect": {
+    "label": "Select the data classification of your form responses [fr]"
```
Comment (anikbrazeau) [inline public/static/locales/fr/form-builder.json:681]:
```suggestion
    "label": "Sélectionnez la classification des données des réponses de votre formulaire"
```

File: public/static/locales/fr/form-builder.json [hunk 6]
```diff
@@ -652,5 +655,32 @@
...
+    },
+    "vaultNote": {
+      "text1": "PROTECTED B class forms must be downloaded from GC Forms. [fr]",
+      "text2": "Email delivery is not available for this classification [fr]"
+    },
+    "saveNote": {
+      "text1": "Save changes to form settings before closing? [fr]",
+      "text2": "Save or Cancel [fr]"
+    },
+    "more": "More [fr]"
```
Comment (anikbrazeau) [inline public/static/locales/fr/form-builder.json:678]:
```suggestion
    "more": "Plus"
```

File: public/static/locales/fr/form-builder.json [hunk 7]
```diff
@@ -652,5 +655,32 @@
...
     },
     "loading": "Chargement..."
+  },
+  "formSettingsModal": {
+    "title": "Form set up [fr]",
+    "okay": "Save [fr]",
+    "cancel": "Cancel [fr]",
+    "classification": "Classification [fr]",
+    "branding": "Branding [fr]",
+    "responseDelivery": "Set up response delivery [fr]",
```
Comment (anikbrazeau) [inline public/static/locales/fr/form-builder.json:665]:
```suggestion
    "responseDelivery": "Configurer la livraison des réponses au formulaire",
```

File: public/static/locales/en/form-builder.json [hunk 8]
```diff
@@ -651,5 +655,32 @@
...
   },
-  "loading": "Loading..."
+  "loading": "Loading...",
+  "formSettingsModal": {
+    "title": "Form set up",
+    "okay": "Save",
+    "cancel": "Cancel",
+    "classification": "Classification",
+    "branding": "Branding",
+    "responseDelivery": "Set up response delivery",
```
Comment (anikbrazeau) [inline public/static/locales/en/form-builder.json:665]:
```suggestion
    "responseDelivery": "Set up form response delivery",
```

File: public/static/locales/en/form-builder.json [hunk 9]
```diff
@@ -651,5 +655,32 @@
...
+    "emailOption": {
+      "label": "Email the responses to an inbox",
+      "note": "-- unavailable for protected B class forms --"
+    },
+    "vaultNote": {
+      "text1": "PROTECTED B class forms must be downloaded from GC Forms.",
```
Comment (anikbrazeau) [inline public/static/locales/en/form-builder.json:671]:
```suggestion
      "text1": "PROTECTED B form responses must be downloaded from GC Forms.",
```

File: public/static/locales/en/form-builder.json [hunk 10]
```diff
@@ -651,5 +655,32 @@
...
+  "loading": "Loading...",
+  "formSettingsModal": {
+    "title": "Form set up",
+    "okay": "Save",
+    "cancel": "Cancel",
+    "classification": "Classification",
+    "branding": "Branding",
+    "responseDelivery": "Set up response delivery",
+    "emailOption": {
+      "label": "Email the responses to an inbox",
```
Comment (anikbrazeau) [inline public/static/locales/en/form-builder.json:667]:
```suggestion
      "label": "Receive responses by email",
```

File: public/static/locales/en/form-builder.json [hunk 11]
```diff
@@ -651,5 +655,32 @@
...
+      "note": "-- unavailable for protected B class forms --"
```
Comment (anikbrazeau) [inline public/static/locales/en/form-builder.json:668]:
```suggestion
      "note": "Not available for PROTECTED B"
```

File: public/static/locales/en/form-builder.json [hunk 12]
```diff
@@ -651,5 +655,32 @@
...
+    },
+    "vaultNote": {
+      "text1": "PROTECTED B class forms must be downloaded from GC Forms.",
+      "text2": "Email delivery is not available for this classification"
```
Comment (anikbrazeau) [inline public/static/locales/en/form-builder.json:672]:
```suggestion
      "text2": "Email delivery is not available for this data classification"
```

File: public/static/locales/en/form-builder.json [hunk 13]
```diff
@@ -651,5 +655,32 @@
...
+    },
+    "saveNote": {
+      "text1": "Save changes to form settings before closing?",
```
Comment (anikbrazeau) [inline public/static/locales/en/form-builder.json:675]:
```suggestion
      "text1": "Save changes before closing?",
```

File: public/static/locales/fr/form-builder.json [hunk 14]
```diff
@@ -652,5 +655,32 @@
...
+    "emailOption": {
+      "label": "Email the responses to an inbox [fr]",
+      "note": "-- unavailable for protected B class forms -- [fr]"
```
Comment (anikbrazeau) [inline public/static/locales/fr/form-builder.json:668]:
```suggestion
      "label": "Recevoir les réponses par courriel",
      "note": "Non disponible pour PROTÉGÉ B"
```

File: public/static/locales/fr/form-builder.json [hunk 15]
```diff
@@ -652,5 +655,32 @@
...
+    },
+    "vaultNote": {
+      "text1": "PROTECTED B class forms must be downloaded from GC Forms. [fr]",
+      "text2": "Email delivery is not available for this classification [fr]"
```
Comment (anikbrazeau) [inline public/static/locales/fr/form-builder.json:672]:
```suggestion
      "text1": "Les réponses aux formulaires PROTÉGÉ B doivent être téléchargées de Formulaires GC",
      "text2": "La livraison des réponses n'est pas disponible pour cette classification de données"
```

File: public/static/locales/en/form-builder.json [hunk 16]
```diff
@@ -651,5 +655,32 @@
       "body2": "unconfirmed responses 35 days overdue. Confirm the receipt of all overdue downloaded responses before saving new ones."
     }
   },
-  "loading": "Loading..."
+  "loading": "Loading...",
+  "formSettingsModal": {
+    "title": "Form set up",
```
Comment (anikbrazeau) [inline public/static/locales/en/form-builder.json:660]:
```suggestion
    "title": "Form set-up",
```

**Comment by timarney** [comment]
@thiessenp-cds FR is in now/

File: public/static/locales/en/form-builder.json [hunk 17]
```diff
@@ -406,8 +406,11 @@
   "publishingRemovesTestResponsesDescription": "Any prior responses will be removed from the system.",
   "publishYourForm": "Publish your form",
   "publishYourFormInstructions": {
-    "text1": "Complete the following items before you publish your",
-    "text2": "form:"
+    "text1": "Complete the following items before you publish your form.",
+    "text2": "",
+    "vaultOption": "— Responses to be downloaded ",
+    "emailOption": "— Responses delivered by email ",
+    "change": "<br>Change settings"
```
Comment (anikbrazeau) [inline public/static/locales/en/form-builder.json:413]:
```suggestion
    "vaultOption": "Responses to be downloaded — ",
    "emailOption": "Responses delivered by email — ",
    "change": "Change settings"
```

File: public/static/locales/fr/form-builder.json [hunk 18]
```diff
@@ -406,8 +406,11 @@
   "publishingRemovesTestResponsesDescription": "Toutes les réponses antérieures seront supprimées du système.",
   "publishYourForm": "Publier votre formulaire",
   "publishYourFormInstructions": {
-    "text1": "Veuillez franchir les étapes suivantes avant de publier votre formulaire",
-    "text2": " :"
+    "text1": "Veuillez franchir les étapes suivantes avant de publier votre formulaire.",
+    "text2": "",
+    "vaultOption": "— Réponses à télécharger",
+    "emailOption": "— Réponses envoyées par courriel",
+    "change": "<br>Modifier ces paramètres"
```
Comment (anikbrazeau) [inline public/static/locales/fr/form-builder.json:413]:
```suggestion
    "vaultOption": "Réponses à télécharger — ",
    "emailOption": "Réponses envoyées par courriel — ",
    "change": "Modifier ces paramètres"
```

**Review by thiessenp-cds — APPROVED** [review]
Nice, LGTM :)

You even remembered the hidden labels!
Screenshot 2023-10-12 at 3 49 03 PM

**Review by thiessenp-cds — APPROVED** [review]
LGTM :)

**Pull request merged**